mkiocccentry.c

```diff
@@ -5452,33 +5452,55 @@ show_registration_url(void)
      * print information about how to register for the IOCCC
      */
     para("",
-	 "To submit to the IOCCC, you must be a registered contestant and have received an",
-	 "IOCCC contest ID (via email) shortly after you've successfully registered.  To do so,",
-	 "please visit:",
+	 "To submit to the IOCCC, you must be a registered contestant",
+         "and have received an IOCCC contest ID (a UUID) and initial",
+         "password (via email). To do so, please visit:",
 	 "",
 	 NULL);
     errno = 0;		/* pre-clear errno for warnp() */
-    ret = fprintf(stderr, "    %s\n", IOCCC_REGISTER_URL);
+    ret = printf("    %s\n", IOCCC_REGISTER_URL);
     if (ret <= 0) {
-	err(190, __func__, "fprintf error printing IOCCC_REGISTER_URL");
+	err(190, __func__, "printf error printing IOCCC_REGISTER_URL");
 	not_reached();
     }
+    para("",
+         "Please also see the FAQ on how to enter the IOCCC:",
+         "",
+         NULL);
+    errno = 0;		/* pre-clear errno for warnp() */
+    ret = printf("    %s\n", IOCCC_REGISTER_FAQ_URL);
+    if (ret <= 0) {
+	err(191, __func__, "printf error printing IOCCC_REGISTER_FAQ_URL");
+	not_reached();
+    }
+    para("",
+        "...and the register, password change and submit help pages:",
+        "",
+        NULL);
+    errno = 0;		/* pre-clear errno for warnp() */
+    ret = printf("    %s\n    %s\n    %s\n", IOCCC_REGISTER_INFO_URL, IOCCC_PW_CHANGE_INFO_URL, IOCCC_SUBMIT_INFO_URL);
+    if (ret <= 0) {
+	err(192, __func__, "printf error printing IOCCC_REGISTER_INFO_URL, IOCCC_PW_CHANGE_INFO_URL and IOCCC_SUBMIT_INFO_URL");
+	not_reached();
+    }
+
 
 #else /* IOCCC_REGISTRATION_READY */
 
     /*
      * print that the registration system is not ready
      */
     para("",
-         "The IOCCC registration process and submit servers are not ready.  You cannot",
-	 "register for the IOCCC or use the IOCCC submit server at this time.",
-	 "Please try again later, and monitor the IOCCC news for updates:",
+         "The IOCCC registration process and submit servers are",
+         "not ready.  You cannot register for the IOCCC or use the",
+         "IOCCC submit server at this time. Please try again later,",
+         "and monitor the IOCCC news for updates:",
```

Comment:
The IOCCC registration process and submit servers are "ready" .. well ready for "pending contest status" and will be complete by the time we are in "open contest status".

Perhaps instead this text should say that they should check the contest status (via the "status.html" page) and if they have registered and the contest status is open, proceed to uploading their submission to the https://submit.ioccc.org server?

Probably earlier, when the tool asks for the username, it could make a reference to registering.

In both places, a link to the FAQ might be helpful.

We will accept this PR as it, but se will should address the above in a subsequent PR, @xexyl

Reply:
I think you forgot .. that's when the macro is undefined. That text is not shown now. Or are you saying that the text itself should be changed to also refer to the status page? But even so the last part will not work as in this place the contest is not open and they cannot submit anything.

So put another way; that text is for when the submit server is not ready (like after IOCCC28 and before IOCCC29 is even contemplated). If you wish me to change the text please give a suggestion and I'll be happy to do that - though I'll be baking soonish.

Reply:
> So put another way: that text is for when the submit server is not ready (like after IOCCC28 and before IOCCC29 is even contemplated). If you wish me to change the text please give a suggestion and I'll be happy to do that - though I'll be baking soonish.

The old text was for when the server didn't exist, and we didn't even know what host name it would run under. 

The server exists now, so the text we identified is not correct and needs to be adjusted. 

Perhaps all that text needs is something as simple as: 

```
When the contest is open, you must upload ((put filename here)) to slot ((put entry number)) on the IOCCC submit server:

  https://submit.ioccc.org

See the FAQ 0.0 at:

  https://www.ioccc.org/faq.html#enter

for more information.
```

Something like that, we kite that in on the iPhone and it probably could be boarded better, but hopefully you'll get the idea 💡.

Reply:
> > So put another way: that text is for when the submit server is not ready (like after IOCCC28 and before IOCCC29 is even contemplated). If you wish me to change the text please give a suggestion and I'll be happy to do that - though I'll be baking soonish.
> 
> The old text was for when the server didn't exist, and we didn't even know what host name it would run under. 
> 
> The server exists now, so the text we identified is not correct and needs to be adjusted. 
> 
> Perhaps all that text needs is something as simple as: 
> 
> ```
> When the contest is open, you must upload ((put filename here)) to slot ((put entry number)) on the IOCCC submit server:
> 
>   https://submit.ioccc.org
> 
> See the FAQ 0.0 at:
> 
>   https://www.ioccc.org/faq.html#enter
> 
> for more information.
> ```
> 
> Something like that, we kite that in on the iPhone and it probably could be boarded better, but hopefully you'll get the general idea 💡, @xexyl. 

In other words: the function should take those args to print out the information to the user? It could be done for when it is open too. 

Is that your idea? If so I can probably do it tomorrow morning. If not please clarify again. 

Thanks and good luck with the merge!

Reply:
> In other words: the function should take those args to print out the information to the user? 

Yes.

> It could be done for when it is open too.

As people might run the command at other times, we can keep the message generic by stating something along the lines of "When the contest is open"

> Is that your idea? If so I can probably do it tomorrow morning. If not please clarify again.

Perhaps:

```
When the contest is open (see https://www.ioccc.org/submit.html), you must upload ((put filename here)) into slot ((put entry number here)) on the IOCCC submit server:

  https://submit.ioccc.org

See the FAQ 0.0 at:

  https://www.ioccc.org/faq.html#enter

for more information.
```

Or something like that.

> Thanks and good luck with the merge!

Backups are now complete.  Proceeding ...

Reply:
> > In other words: the function should take those args to print out the information to the user?
> 
> Yes.
> 
> > It could be done for when it is open too.
> 
> As people might run the command at other times, we can keep the message generic by stating something along the lines of "When the contest is open"
> 
> > Is that your idea? If so I can probably do it tomorrow morning. If not please clarify again.
> 
> Perhaps:
> 
> ```
> When the contest is open (see https://www.ioccc.org/submit.html), you must upload ((put filename here)) into slot ((put entry number here)) on the IOCCC submit server:
> 
>   https://submit.ioccc.org
> 
> See the FAQ 0.0 at:
> 
>   https://www.ioccc.org/faq.html#enter
> 
> for more information.
> ```
> 
> Or something like that.
> 
> > Thanks and good luck with the merge!
> 
> Backups are now complete. Proceeding ...

Thanks for clarifying and good luck! I have to go now but I'll do this tomorrow morning. If I don't get to it please do remind me. Hopefully you get the merge done today and we can also test links! I think it should be good but for now I bid you a good day!

Reply:
I see you made a new release. Does this mean that you took care of this (can't check now - or I can check but I won't be able to tell yet as I'm still bleary eyed) or are you just preparing for it for when I do it? Please advise. Hope you got some sleep and good job on the great fork merge!!

Reply:
> I see you made a new release. Does this mean that you took care of this (can't check now - or I can check but I won't be able to tell yet as I'm still bleary eyed) or are you just preparing for it for when I do it? Please advise. Hope you got some sleep and good job on the great fork merge!!

I see you did not make the update ... however since you made a new release I do not know how you wish me to update it. Please advise and I'd be happy to do it for you.

Reply:
We had to make a new release as part of the **Great Fork Merge** without the mod.  However, since we are now in pending mode, we can modify the code.

Please form a PR and prep for a new release to fix it, @xexyl.

Reply:
> We had to make a new release as part of the **Great Fork Merge** without the mod. However, since we are now in pending mode, we can modify the code.
> 
> Please form a PR and prep for a new release to fix it, @xexyl.
> 
> ## UPDATE 0
> We had to move forward as it was and push the site update out the door. As it was were up until 02:00 getting it out.
> 
> So we have a month in pending state, before things go into open, where things like this repo can be re-released. So please do your PR and prep for a 2.1.1 release .. let's not wait until January to do this. 🍪🤓

Sure .. I do have to open those new issues too unfortunately. I think I am out of energy for the day with the contest but I hope to at least open the new issues: in a bit. I have to do something else first. However: before I do go I want you to clarify something that I'm unsure of.

This new text. Is that when the submit server is ready (that macro is defined) or when it's not (macro undefined)? Or do you want the macro to no longer be there? Please advise and then hopefully tomorrow I can get to this.

If I don't please remind me. I have to go for a while but when back I hope to open those new issues.